tests/ed.fs

```diff
@@ -195,10 +195,10 @@ create 'saved-output  1000 allot
 \ assembler macro push-a
 : save-output ( c -- ) 
    [ push-a ]  \ "dex dex  sta 0,x  stz 1,x" - push A to TOS
-   [ phy 36 lda.z pha  37 lda.z pha ] \ Save y and tmp1.
+   [ phy 34 lda.z pha  35 lda.z pha ] \ Save y and tmp1.
```

Comment:
eek.

Reply:
That's a good catch (and a bit of a mess).
This could be fixed by writing the rest of this routine in assembly.  The issue is that this is an I/O word (used for I/O redirection), which is not allowed to modify any of the registers.  The pushing of tmp1 is because at least one of the Forth words below uses it.

Your band-aid is OK for right now, but we should probably fix this.  Do you want me to rewrite it or do you want to do it?

Reply:
you'll probably be much more efficient here than me

Reply:
That's fine.  I'll wait until this PR is merged before merging my solution that doesn't use tmp1 at all.